packages/client/src/workflow-client.ts

```diff
@@ -946,6 +926,31 @@ export class WorkflowClient {
       followRuns: options?.followRuns ?? true,
     });
   }
+
+  /**
+   * List workflows by given `query`.
+   *
+   * See more info on how to build the query in the Temporal docs: https://docs.temporal.io/visibility
+   *
+   * This API requires Elasticsearch based visibility as of server version 1.18.
+   */
+  public async *list(query: string, options?: ListOptions): AsyncIterable<WorkflowExecutionInfo> {
```

Comment:
If describe is interceptable, this should be right?

Reply:
I don't see any point making either of them interceptable.

Reply:
I support either approach, but the one-is-one-isn't seems a little strange. Today describe is interceptable.

Reply:
Tracking this in a separate issue.
I want to clean up client interceptors a bit before adding this.